test/units/assert.sh

```diff
@@ -3,42 +3,56 @@
 
 # utility functions for shell tests
 
-assert_true() {
+assert_true() {(
     local rc
 
-    set +e
+    set +ex
+
     "$@"
     rc=$?
-    set -e
     if [[ "$rc" != "0" ]]; then
         echo "FAIL: command '$*' failed with exit code $rc" >&2
         exit 1
     fi
-}
+)}
+
 
+assert_eq() {(
+    set +ex
 
-assert_eq() {
     if [[ "$1" != "$2" ]]; then
```

Comment:
```suggestion
    if [[ "${1?}" != "${2?}" ]]; then
```
(No colons, since we want fail only on unset values, not empty ones)